packages/components/src/input-control/test/index.js

```diff
@@ -41,18 +41,6 @@ describe( 'InputControl', () => {
 		} );
 	} );
 
-	describe( 'Ensurance of focus for number inputs', () => {
-		it( 'should focus its input on mousedown events', () => {
```

Comment:
Maybe a silly question, but — shouldn't we keep this unit test?

Technically the code in this PR is not necessary, given change in #40518 — and therefore I would keep the unit tests as they are, and expect the to still pass.

Reply:
The test was added along with the code in #29305. I expected such a test to never fail https://github.com/WordPress/gutenberg/pull/29305#issuecomment-785600450 because inputs are focused on `mousedown` (except Firefox when `mousedown` is on the spin buttons). Apparently that focus is not synchronous. So without the code this test fails.

Reply:
I see, thank you for the context!

I tried using `user-event` and it seems to fix the issue:

```js
	describe( 'Ensurance of focus for number inputs', () => {
		it( 'should focus its input on mousedown events', async () => {
			const user = userEvent.setup( {
				advanceTimers: jest.advanceTimersByTime,
			} );

			const spy = jest.fn();
			render( <InputControl type="number" onFocus={ spy } /> );

			const input = getInput();
			await user.click( input );

			expect( spy ).toHaveBeenCalledTimes( 1 );
		} );
	} );
```